financial-connections/src/main/java/com/stripe/android/financialconnections/features/MarkdownParser.kt

```diff
@@ -1,74 +1,21 @@
 package com.stripe.android.financialconnections.features
 
-import com.stripe.android.financialconnections.model.Bullet
-import com.stripe.android.financialconnections.model.ConsentPane
-import com.stripe.android.financialconnections.model.ConsentPaneBody
-import com.stripe.android.financialconnections.model.DataAccessNotice
-import com.stripe.android.financialconnections.model.DataAccessNoticeBody
-import com.stripe.android.financialconnections.model.LegalDetailsBody
-import com.stripe.android.financialconnections.model.LegalDetailsNotice
-
 internal object MarkdownParser {
-    private val markDownToHtmlRegex = listOf(
+
+    private val markDownToHtmlRegex = listOf<Pair<Regex, (MatchResult) -> CharSequence>>(
         // bold, italics rules
-        "\\*\\*\\s?([^\\n]+)\\*\\*".toRegex() to "<b>$1</b>",
-        "__([^_]+)__".toRegex() to "<b>$1</b>",
+        "\\*\\*(.*?)\\*\\*".toRegex() to { "<b>${it.groupValues[1]}</b>" },
+        "__([^_]+)__".toRegex() to { "<b>${it.groupValues[1]}</b>" },
 
         // links
-        "\\[([^]]+)]\\(([^)]+)\\)".toRegex() to "<a href=\"$2\">$1</a>"
+        "\\[([^]]+)]\\(([^)]+)\\)".toRegex() to { "<a href=\"${it.groupValues[2]}\">${it.groupValues[1]}</a>" }
     )
 
     internal fun toHtml(string: String): String {
         var newText = string
         markDownToHtmlRegex.forEach { (regex, replacement) ->
-            newText = newText.replace(regex, replacement)
+            newText = regex.replace(newText, replacement)
```

Comment:
is this expected?

Reply:
it is expected yes. We're now using `Regex#replace` instead of `String#replace` so that we can access `groupValues` which are more clear)